woocommerce/woocommerce-android
# Bundle quantity rules validation

Part of: #9541 

### Why
As part of the support for extensions milestone 2, we are working on support bundle products in the order creation flow.

### Description
This PR validates the product quantity rules before saving the bundle configuration. To achieve this, first, we fetch the bundle quantity rules from the metadata and save it in the bundle rules. We use these rules later in the product configuration screen to check if the current configuration is valid. If the configuration is not valid, we show a message at the top of the screen and disable the Save configuration button. We also use the bundle rules to check the min-max rules for products inside the bundle. Now, the stepper control controlling the quantity of products inside the bundle will match the inner product quantity rules.

### Testing instructions

#### Prerequisites
1. Install the product bundles extension
2. Create different product bundles containing quantity rules (for the bundle and inner products)

#### Testing
1. Open the orders tab
2. Tap on create new order (+)
3. Tap on add products
4. Select a bundle product containing quantity rules
5. Play with the bundle configuration
6. Check that when the rules are not met, a message appears at the top of the screen
7. Check that when the rules are not met, the Save configuration button is disabled
8. Check that the stepper control for inner products min-max matches the inner product min-max rules
9. Check that when the rules are met, there is no message at the top of the screen
10. Check that when the rules are met, the Save configuration button is enabled

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/3693e096-eea3-48ca-8f49-edaf7c19f444
